x-pack/plugins/apm/server/lib/helpers/setup_request.test.ts

```diff
@@ -117,102 +125,96 @@ describe('setupRequest', () => {
+      const { apmEventClient } = await setupRequest(mockContext, mockRequest);
+      await apmEventClient.search({
+        apm: {
+          events: [ProcessorEvent.transaction],
+        },
+        body: { query: { bool: { filter: [{ term: 'someTerm' }] } } },
       });
-
-      it('should add `observer.version_major` filter if none exists', async () => {
-        const { mockContext, mockRequest } = getMockRequest();
-        const { client } = await setupRequest(mockContext, mockRequest);
-        await client.search({ index: 'apm-*' });
-        const params =
-          mockContext.core.elasticsearch.legacy.client.callAsCurrentUser.mock
-            .calls[0][1];
-        expect(params.body).toEqual({
-          query: {
-            bool: {
-              filter: [{ range: { 'observer.version_major': { gte: 7 } } }],
-            },
+      const params =
+        mockContext.core.elasticsearch.legacy.client.callAsCurrentUser.mock
+          .calls[0][1];
+      expect(params.body).toEqual({
+        query: {
+          bool: {
+            filter: [
+              { term: 'someTerm' },
+              { terms: { [PROCESSOR_EVENT]: ['transaction'] } },
+              { range: { 'observer.version_major': { gte: 7 } } },
+            ],
           },
-        });
+        },
       });
+    });
 
-      it('should not add `observer.version_major` filter if `includeLegacyData=true`', async () => {
-        const { mockContext, mockRequest } = getMockRequest();
-        const { client } = await setupRequest(mockContext, mockRequest);
-        await client.search(
-          {
-            index: 'apm-*',
-            body: { query: { bool: { filter: [{ term: 'someTerm' }] } } },
+    it('should add `observer.version_major` filter if none exists', async () => {
+      const { mockContext, mockRequest } = getMockRequest();
+      const { apmEventClient } = await setupRequest(mockContext, mockRequest);
+      await apmEventClient.search({
+        apm: {
+          events: [ProcessorEvent.error],
+        },
+      });
+      const params =
+        mockContext.core.elasticsearch.legacy.client.callAsCurrentUser.mock
+          .calls[0][1];
+      expect(params.body).toEqual({
+        query: {
+          bool: {
+            filter: [
+              { terms: { [PROCESSOR_EVENT]: ['error'] } },
+              { range: { 'observer.version_major': { gte: 7 } } },
+            ],
           },
-          {
-            includeLegacyData: true,
-          }
-        );
-        const params =
-          mockContext.core.elasticsearch.legacy.client.callAsCurrentUser.mock
-            .calls[0][1];
-        expect(params.body).toEqual({
-          query: { bool: { filter: [{ term: 'someTerm' }] } },
-        });
+        },
       });
     });
 
-    it('if index is not an APM index, it should not add `observer.version_major` filter', async () => {
+    it('should not add `observer.version_major` filter if `includeLegacyData=true`', async () => {
```

Comment:
This would read better as `describe('when includeLegacyData=true', () => { it('does not add an observer.version_major filter', () => ...`

Reply:
Agreed, restructured the tests.